### ISSUE TITLE
Use records instead of position arguments for variables

### ISSUE THREAD

**Issue by gyzerok** [jahewson/elm-graphql#4]
What do you think about it?

So instead of
```elm
StarWars.queryFriends "1000"
```
you would have
```elm
StarWars.queryFriends { id = "1000" }
```
I guess it would be more maintainable in case you have a lot different variables.

I can try to introduce corresponding PR.

**Comment by jahewson** [comment]
I had wondered about that too. Facebook's `graphql-js` library uses the following format: 

```js
let arguments = { someId: "1234" }
graphql(schema, query, null, arguments)
```

So I think that the answer is yes, we should do this.

**Comment by note89** [comment]
another benefit of this except beeing much simpler is that it would probably solve this error i got 
```
query queryLinks($query: String!){
  store{
    linkConnection(query: $query){
      edges{
        node{
          id,
          title,
          url
        }
      }
    }
  }
}
```

where the name `query` overlaps with one of the variables in the query function. 

i just changed the name for now. 
`$query -> $queryParam`

**Comment by jahewson** [comment]
I've fixed the naming collision issue in 9107e70da83294c7e7169022a40f6f205e70057e.

**Comment by jahewson** [comment]
Ok, we now have records for parameters in b0626103d98307938c9099c4547017178525c747
